src/core/js/models/configModel.js

```diff
@@ -16,13 +16,34 @@ export default class ConfigModel extends LockingModel {
     };
   }
 
+  /**
+   * Parse the incoming search queries for a 'lang' parameter value to return
+   */
+  setValuesFromURLParams() {
+    const paramMappings = {
+      lang: '_defaultLanguage'
+    };
+
+    const params = new URLSearchParams(window.location.search);
+
+    for (const [key, value] of Object.entries(paramMappings)) {
+      const passedVal = params.get(key);
+
+      if (passedVal) {
+        this.set(value, passedVal);
+      }
+    }
```

Comment:
```suggestion
    const paramMappings = {
      dir: '_defaultDirection',
      lang: '_defaultLanguage'
    };

    const params = new URLSearchParams(window.location.search);

    for (const [key, value] of Object.entries(paramMappings)) {
      const passedVal = params.get(key);
      if (!passedVal) continue;
      
      this.set(value, passedVal);
    }
```

Reply:
not quite sure why every single line is being flagged with a change...! but i'm sure you get the general idea anyway